homeassistant/helpers/entity_component.py

```diff
@@ -74,34 +74,30 @@ def add_entities(self, new_entities):
 
                 entity.update_ha_state()
 
-        if self.group is None and self.group_name is not None:
-            self.group = group.Group(self.hass, self.group_name,
-                                     user_defined=False)
-
         if self.group is not None:
             self.group.update_tracked_entity_ids(self.entities.keys())
 
-        self._start_polling()
-
     def extract_from_service(self, service):
         """
         Takes a service and extracts all known entities.
         Will return all if no entity IDs given in service.
         """
+        entities = self.get_entities()
+
         if ATTR_ENTITY_ID not in service.data:
-            return self.entities.values()
+            return entities.values()
         else:
-            return [self.entities[entity_id] for entity_id
+            return [entities[entity_id] for entity_id
                     in extract_entity_ids(self.hass, service)
-                    if entity_id in self.entities]
+                    if entity_id in entities]
 
-    def _update_entity_states(self, now):
-        """ Update the states of all the entities. """
-        self.logger.info("Updating %s entities", self.domain)
+    def get_entities(self):
+        """ Returns all the platforms and component entities """
+        entities = self.entities.copy()
```

Comment:
There should be only 1 source of truth. We either keep all entities spread out over the platforms or we keep them in a hash on the EntityComponent.